[feat ]Add CoverageChecker to setInterruptHandler in util/Promise.scala

Zodbigt opened the pull request:
This PR adds coverage checking has been added to the method 'setInterruptHandler' in util/Promise.scala. Resolves #16 

[Issue: #16]

Zodbigt committed:
[feat] Add CoverageChecker to setInterruptHandler

Coverage checking has been added to the method 'setInterruptHandler' in util/Promise.scala

[Issue: #16]

Zodbigt committed:
[fix] Print test statment for setInterruptHandler

This commit makes print stament unique and visable

[issue: #16]

stevensdavid commented on util-core/src/test/scala/com/twitter/util/DurationTest.scala:
```diff
@@ -296,7 +296,7 @@ class DurationTest extends { val ops: Duration.type = Duration } with TimeLikeSp
   "CoverageChecker" should {
     "print map" in {
-      println(CoverageChecker.map.getOrElse("parse", sys.error(s"unexpected key")).mkString("[",", ", "]"))
+      println("[Coverage - parse] - " + CoverageChecker.map.getOrElse("parse", sys.error(s"unexpected key")).mkString("[",", ", "]"))
```
this shouldn't be here!

Zodbigt replied:
Ill fix that!

stevensdavid commented on util-core/src/main/scala/com/twitter/util/Promise.scala:
```diff
@@ -509,25 +509,55 @@ class Promise[A] extends Future[A] with Promise.Responder[A] with Updatable[Try[
+        CoverageChecker.reached(funcName, 8)
+        p.setInterruptHandler(f)
+      }
+    }
```
I guess that technically there's a default case `case _: => ()` that should be tracked?

felixlut replied:
No there shouldn't be one. See issue #27

Zodbigt committed:
[fix] Removed change that belonging to an other issue.

This commit changes back to orginal code for DuratinTest.scala, since it belongs to an other issue.

[issue: #16]

stevensdavid committed:
Merge branch 'code-coverage' into issue/16
